cartographer_ros/src/cartographer_node_main.cc

```diff
@@ -456,6 +466,49 @@ bool Node::HandleSubmapQuery(
   return true;
 }
 
+bool Node::HandleFinishTrajectory(
+    ::cartographer_ros_msgs::FinishTrajectory::Request& request,
+    ::cartographer_ros_msgs::FinishTrajectory::Response& response) {
+  carto::common::MutexLocker lock(&mutex_);
+  // TODO(whess): Add multi-trajectory support.
+  sensor_collator_.FinishTrajectory(kTrajectoryBuilderId);
```

Comment:
Have you tested what happens if this is called while sensor data is still coming in? Looking at the code, I would expect any data that arrives after this process begins to be queued up by ROS and then ignored by the collator.

Would be nice to verify that though. A crash would be bad :)